tests/general/ArchiveOrgBearTest.py

```diff
@@ -0,0 +1,322 @@
+            'https://www.google.com/',
+            headers={
+                'Link': '<https://www.google.com/>; rel="origi'
+                        'nal", <http://web.archive.org/web/tim'
+                        'emap/link/https://www.google.com/>; r'
+                        'el="timemap"; type="application/link-'
+                        'format", <http://web.archive.org/web/'
+                        'https://www.google.com/>; rel="timega'
+                        'te", <http://web.archive.org/web/1998'
+                        '1111184551/http://google.com:80/>; re'
+                        'l="first memento"; datetime="Wed, 11 '
+                        'Nov 1998 18:45:51 GMT", <http://web.a'
+                        'rchive.org/web/20170420064646/http://'
+                        'www.google.com/>; rel="prev memento";'
+                        ' datetime="Thu, 20 Apr 2017 06:46:46 '
+                        'GMT", <http://web.archive.org/web/201'
+                        '70421070937/https://www.google.com/>;'
+                        ' rel="memento"; datetime="Fri, 21 Apr'
+                        ' 2017 07:09:37 GMT", <http://web.arch'
+                        'ive.org/web/20170422071758/https://ww'
+                        'w.google.com/>; rel="next memento"; d'
+                        'atetime="Sat, 22 Apr 2017 07:17:58 GM'
+                        'T", <http://web.archive.org/web/19981'
+                        '111184551/http://google.com:80/>; rel'
+                        '="last memento"; datetime="Wed, 11 No'
+                        'v 1998 18:45:51 GMT"'
+            },
+            status_code=200)
+    else:
+        m.head(
+            'http://timetravel.mementoweb.org/timegate/%s' % url,
+            headers={})
+
+
+def generate_redirects(m, url, redirect_length=1):
+    """
+    Generate redirects path for the given url.
+
+    :param m:               A `request_mock.Mocker()` instance.
+    :param url:             The ``url`` to be redirected.
+    :param redirect_length: How many redirects will be generated.
+    """
+    m.head(
+        url,
+        headers={'Location': '%s/1' % url},
+        status_code=302)
+
+    if redirect_length > 1:
+        for i in range(redirect_length - 1):
+            m.head(
+                '%s/%s' % (url, str(i + 1)),
+                headers={'Location': '%s/%s' % (url, str(i + 2))},
+                status_code=302
+            )
+
+        m.head('%s/%s' % (url, str(redirect_length)),
+               status_code=200)
+    else:
+        m.head('%s/1' % url, status_code=200)
+
+
+class ArchiveOrgBearTest(unittest.TestCase):
+    """
+    The tests are mocked (don't actually connect to internet) and
+    return the int conversion of the last three chars of
+    the URL as status code.
+
+    Check ``custom matcher`` for more info on implementation.
+    """
+
+    def assertResult(self, valid_file=None, invalid_file=None, settings={}):
```

Comment:
I think we have the functionality you're building here in LocalBearTestHelper or so